server/graphql/v2/mutation/MemberMutations.js

```diff
@@ -40,6 +47,231 @@ const memberMutations = {
+      }
+
+      const memberParams = {
+        ...pick(args, ['role', 'description', 'since']),
+        MemberCollectiveId: memberAccount.id,
+        CreatedByUserId: req.remoteUser.id,
+      };
+
+      // Invite member
+      await models.MemberInvitation.invite(account, memberParams);
+
+      const invitation = await models.MemberInvitation.findOne({
+        where: {
+          CollectiveId: account.id,
+          MemberCollectiveId: memberParams.MemberCollectiveId,
+        },
+      });
+
+      return invitation;
+    },
+  },
+  editMember: {
+    type: Member,
+    description: 'Invite a new member to the Collective',
+    args: {
+      memberAccount: {
+        type: GraphQLNonNull(AccountReferenceInput),
+        description: 'Reference to an account for the donating Collective',
+      },
+      account: {
+        type: GraphQLNonNull(AccountReferenceInput),
+        description: 'Reference to an account for the receiving Collective',
+      },
+      role: {
+        type: MemberRole,
+        description: 'Role of member',
+      },
+      description: {
+        type: GraphQLString,
+      },
+      since: {
+        type: ISODateTime,
+      },
+      isInvitation: {
+        type: GraphQLBoolean,
+      },
+    },
+    async resolve(_, args, req) {
+      if (!req.remoteUser) {
+        throw new Unauthorized('You need to be logged in to invite a member.');
+      }
+
+      let { memberAccount, account } = args;
+
+      memberAccount = await fetchAccountWithReference(memberAccount);
+      account = await fetchAccountWithReference(account);
+
+      if (!req.remoteUser.isAdminOfCollective(account)) {
+        throw new Unauthorized('Only admins can edit members.');
+      }
+
+      if (![MemberRoles.ACCOUNTANT, MemberRoles.ADMIN, MemberRoles.MEMBER].includes(args.role)) {
+        throw new Forbidden('You can only edit accountants, admins, or members.');
+      }
+
+      if (args.role !== MemberRoles.ADMIN) {
+        const admins = await models.Member.findAll({
+          where: {
+            CollectiveId: account.id,
+            role: MemberRoles.ADMIN,
+          },
+        });
+
+        if (admins.length === 1) {
+          if (admins[0].MemberCollectiveId === memberAccount.id) {
+            throw new Forbidden('There must be at least one admin for the account.');
+          }
+        }
+      }
+
+      // Edit member
+      let member;
+      const editableAttributes = pick(args, ['role', 'description', 'since']);
+
+      if (args.isInvitation) {
+        await models.MemberInvitation.update(editableAttributes, {
+          where: {
+            MemberCollectiveId: memberAccount.id,
+            CollectiveId: account.id,
+          },
+        });
+
+        member = await models.Member.findOne({
+          where: {
+            CollectiveId: account.id,
+            MemberCollectiveId: memberAccount.id,
+          },
+        });
```

Comment:
This won't return anything, since at this point there's only an invitation and no `Members` entry. I don't think unions are supported by GraphQL at the moment, so the simplest might be to have two mutations: `editMember` and `editMemberInvitiation`